src/clj/puppetlabs/http/client/sync.clj

```diff
@@ -23,43 +23,63 @@
   (select-keys opts [:url :method :headers :body :decompress-body :as :query-params]))
 
 (defn request-with-client
-  [req client]
-  (let [{:keys [error] :as resp} @(async/request-with-client req nil client)]
+  [req client metric-registry]
+  (let [{:keys [error] :as resp} @(async/request-with-client req nil client metric-registry)]
     (if error
       (throw error)
       resp)))
 
+(defn get-client-metrics
+  ([metric-registry]
+   (async/get-client-metrics metric-registry))
+  ([metric-registry metric-filter]
+   (async/get-client-metrics metric-registry metric-filter)))
+
+(defn get-client-metrics-data
+  ([metric-registry]
+   (async/get-client-metrics-data metric-registry))
+  ([metric-registry metric-filter]
+   (async/get-client-metrics-data metric-registry metric-filter)))
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
 (defn request
   [req]
   (with-open [client (async/create-default-client (extract-client-opts req))]
-    (request-with-client (extract-request-opts req) client)))
+    (request-with-client (extract-request-opts req) client nil)))
```

Comment:
Seems like it would be good to have a way to pass in a `MetricsRegistry` even for one-off requests, not just for cases where you construct a full client?  To do that, we might need to have the client pass in the `MetricsRegistry` via an option in the req map?